### PR TITLE
added envFrom option to vmagent deployment and statefulset

### PR DESCRIPTION
**Pull request by neinkob15**

### CODE REVIEW

**Comment by tenmozes** [comment]
@neinkob15 could you please remove empty valyes.yml file from your PR

**Comment by tenmozes** [comment]
@Haleygo, could you please take a look? Should we add the same functionality to other charts if everything is okay?

**Comment by neinkob15** [comment]
@tenmozes empty file was removed

File: charts/victoria-metrics-agent/values.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ deployment:
 
 # ref:  https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/
 statefulset:
-  enabled: false
+  enabled: true
```
Comment (Haleygo) [inline charts/victoria-metrics-agent/values.yaml:19]: Please revert this value as it brings breaking change.

**Comment by Haleygo** [comment]
> @Haleygo, could you please take a look? Should we add the same functionality to other charts if everything is okay?

yeah, I think it's ok to add `envFrom` in all the charts.

File: charts/victoria-metrics-agent/values.yaml (hunk 1 above)
Comment (neinkob15) [reply charts/victoria-metrics-agent/values.yaml]: You are totally right, I forgot to reset that one. Is now at false again

**Comment by neinkob15** [comment]
@tenmozes @Haleygo I added the envFrom option to all the charts now. Please take a look at my changes

File: charts/victoria-metrics-anomaly/templates/deployment.yaml [hunk 2]
```diff
@@ -43,6 +43,15 @@ spec:
           {{- with (include "chart.license.flag" .) }}
             - {{ . }}
           {{- end}}
+          {{- if .Values.envFrom }}
+          envFrom:
+            {{- with .Values.envFrom -}}
+            {{ toYaml . | nindent 12 }}
+            {{- end }}
+          {{- end }}
+          {{- with .Values.env }}
```
Comment (Haleygo) [inline charts/victoria-metrics-anomaly/templates/deployment.yaml:52]: good catch!

File: charts/victoria-metrics-gateway/README.md [hunk 3]
```diff
@@ -8,12 +8,15 @@ Victoria Metrics Gateway - Auth & Rate-Limitting proxy for Victoria Metrics
...
-* [How to Uninstall](#how-to-uninstall)
-* [How to use JWT signature verification](#how-to-use-jwt-signature-verification)
-* [Documentation of Helm Chart](#documentation-of-helm-chart)
+- [Victoria Metrics Helm Chart for vmgateway](#victoria-metrics-helm-chart-for-vmgateway)
+- [Table of Content](#table-of-content)
+  - [Prerequisites](#prerequisites)
+  - [Chart Details](#chart-details)
+  - [How to install](#how-to-install)
+  - [How to uninstall](#how-to-uninstall)
+- [How to use JWT signature verification](#how-to-use-jwt-signature-verification)
```
Comment (Haleygo) [inline charts/victoria-metrics-gateway/README.md:17]: Is there particular reason to change this index?

Comment (neinkob15) [reply charts/victoria-metrics-gateway/README.md]: No, you're right, that was my VS Code extension to auto-generate the table of contents on saving the file.
I reverted that change

**Review by Haleygo — APPROVED** [review]
LGTM!
Thanks for contribution!

**Pull request merged**